Source/JavaScriptCore/llint/LowLevelInterpreter.asm

```diff
@@ -1603,16 +1648,63 @@ end
 # EncodedJSValue vmEntryToJavaScript(void* code, VM* vm, ProtoCallFrame* protoFrame)
 # EncodedJSValue vmEntryToNativeFunction(void* code, VM* vm, ProtoCallFrame* protoFrame)
 
+macro frameForCalleeSaveVerification()
+    if ARM64 or ARM64E
+        const scratch = t9
+    else
+        const scratch = t5
+    end
+
+    # Don't do this always since it screws up btjs.
+    loadBoolJSCOption(validateVMEntryCalleeSaves, scratch)
+    btbz scratch, .continue
+
+    functionPrologue()
+    # This VMEntryRecord is used to record what our callee saves were originally so we can check we didn't screw up somewhere.
+    vmEntryRecord(cfr, sp)
+    checkStackPointerAlignment(scratch, 0xbad0dc02)
+    move cfr, scratch
+    copyCalleeSavesToEntryFrameCalleeSavesBuffer(scratch)
+
+    call .continue
+
+    vmEntryRecord(cfr, scratch)
+    leap VMEntryRecord::calleeSaveRegistersBuffer[scratch], scratch
+    forEachGPCalleeSave(macro (reg, index)
+        bqeq index * MachineRegisterSize[scratch], reg, .ok
+        break
+    .ok:
+    end)
+    forEachFPCalleeSave(macro (reg, index)
+        fd2q reg, t3
+        bqeq (index + (constexpr GPRInfo::numberOfCalleeSaveRegisters)) * MachineRegisterSize[scratch], t3, .ok
+        break
+    .ok:
+    end)
+
+    move cfr, sp
+    functionEpilogue()
+    ret
+.continue:
+end
+
 if C_LOOP
     _llint_vm_entry_to_javascript:
 else
     global _vmEntryToJavaScript
     _vmEntryToJavaScript:
+
+    if ASSERT_ENABLED and (ARM64 or ARM64E or X86_64)
+        frameForCalleeSaveVerification()
+    end
```

Comment:
nit:
1. Let's rename `frameForCalleeSaveVerification` to `injectCalleeSaveVerificationFrameIfNeeded`.
2. Let's do the `if` condition checks inside the macro instead of at all the client locations.